python/amici/sbml_import.py

```diff
@@ -1108,12 +1181,20 @@ def writeFunctionFile(self, function, assume_pow_positivity):
 
         lines.append('')
 
-        lines.append('void ' + function + '_' + self.modelName + signature + '{')
+        lines.append('void ' + function + '_' +
+                     self.modelName + signature + '{')
 
         # function body
         body = self.getFunctionBody(function)
-        if assume_pow_positivity:
-            body = [re.sub(r'(^|\W)pow\(', r'\1amici::pos_pow(', line) for line in body]
+        if assume_pow_positivity \
+                and 'assume_pow_positivity' in self.functions[function].keys()\
+                and self.functions[function]['assume_pow_positivity']:
+            body = [re.sub(r'(^|\W)pow\(', r'\1amici::pos_pow(', line) for line
+                    in body]
+            # execute this twice to catch cases where the ending ( would be the
+            # starting (^|\W) for the following match
```

Comment:
This will fail for 'pow(pow(pow(` etc.. Maybe directly go for:
```
while True:
        b = re.sub(r'(^|\W)pow\(', r'\1amici::pos_pow(', a)
        if b == a: break 
        else: a = b
```

Reply:
Don't you think that 2 passes will always be fine?

`re.sub(r'(^|\W)pow\(', r'\1amici::pos_pow(', 'pow(pow(pow(')` yields `'amici::pos_pow(pow(amici::pos_pow('`

`re.sub(r'(^|\W)pow\(', r'\1amici::pos_pow(',re.sub(r'(^|\W)pow\(', r'\1amici::pos_pow(', 'pow(pow(pow('))` yields `'amici::pos_pow(amici::pos_pow(amici::pos_pow('`

Reply:
Oh right, it always matches every second and thus should work for any number. So it's fine.